redis/src/caching/config.rs

```diff
@@ -0,0 +1,49 @@
+use std::num::NonZeroUsize;
+use std::time::Duration;
+
+/// Defines CacheMode of CacheConfig
+#[derive(Clone, PartialEq, Debug, Copy)]
+pub enum CacheMode {
+    /// Don't cache
+    None,
+    /// Try to cache every command.
+    All,
+    /// Only try to cache commands which explicitly called cache methods.
+    OptIn,
+}
+
+/// Holds configuration to be used by [`crate::caching::CacheManager`]
+#[derive(Clone, Debug, Copy)]
+pub struct CacheConfig {
```

Comment:
please use the builder pattern instead of exposing internal fields. That way we can refactor the class without breaking user code.

Reply:
I've stopped exposing internal field and added setter methods.